preprocessy/handlenullvalues/_main.py

```diff
@@ -81,37 +99,47 @@ def __validate_input(self):
                 )
             elif type(self.fill_values) is not dict:
                 raise TypeError(f'Expected dict value for argument "fill_values" ')
-            column_list = list(self.df)
+            col_list = list(self.df)
             user_column_list = list(self.fill_values.keys())
             for column in user_column_list:
-                if column not in column_list:
+                if column not in col_list:
                     raise ArgumentsError(f"Column {column} does not exist in dataframe")
 
+        if self.column_list is not None:
+            if not isinstance(self.column_list,list):
+                raise TypeError(f"Expected List for argument \"column_list\"")
+            if self.drop is not None and self.drop and len(self.column_list) == 0:
+                warnings.warn("\"column_list\" is empty, no columns will be dropped. If you want to drop rows, do not pass \"column_list\" in the arguments or pass None",UserWarning)
+            if len(self.column_list) != 0:
+                cols=list(self.df)
```

Comment:
`list(self.df)` is being recomputed here. You can use `col_list` defined earlier.